modules/hal_ethos_u/Kconfig

```diff
@@ -40,6 +40,32 @@ config ARM_ETHOS_U_NPU_NAME
 	help
 		Name of the used Arm NPU
 
-module = ARM_ETHOS_U
-module-str = arm_ethos_u
-source "subsys/logging/Kconfig.template.log_config"
+choice "ARM_ETHOS_U_LOG_LEVEL_CHOICE"
+	prompt "Max compiled-in log level for arm_ethos_u"
+	default ARM_ETHOS_U_LOG_LEVEL_WRN
+	depends on STDOUT_CONSOLE
+
+config ARM_ETHOS_U_LOG_LEVEL_ERR
+	bool "Error"
+
+config ARM_ETHOS_U_LOG_LEVEL_WRN
+	bool "Warning"
+
+config ARM_ETHOS_U_LOG_LEVEL_INF
+	bool "Info"
+
+config ARM_ETHOS_U_LOG_LEVEL_DBG
+	bool "Debug"
+
+config ARM_ETHOS_U_LOG_LEVEL_DEFAULT
+	bool "Default"
+
+endchoice
+
+config ARM_ETHOS_U_LOG_LEVEL
+	int
+	depends on STDOUT_CONSOLE
+	default 1 if ARM_ETHOS_U_LOG_LEVEL_ERR
+	default 2 if ARM_ETHOS_U_LOG_LEVEL_WRN
+	default 3 if ARM_ETHOS_U_LOG_LEVEL_INF
+	default 4 if ARM_ETHOS_U_LOG_LEVEL_DBG
```

Comment:
Wow, this is rather confusing.  For anyone else coming in on the review unfamiliar with the ethos HAL, the secret sauce for mapping error levels here is https://github.com/zephyrproject-rtos/zephyr/blob/1fc16e65656e742fbe0adc89e0fdf685c51525a5/modules/hal_ethos_u/CMakeLists.txt#L9

Reply:
Sounds like this may deserve a comment in this file.